src/python/turicreate/test/test_activity_classifier.py

```diff
@@ -150,6 +150,26 @@ def test_create_missing_validation_set(self):
                 validation_set=sf,
             )
 
+    def test_create_single_input_column(self):
+        sf_label = random.randint(0, self.num_labels - 1)
+        sf_session_id = max(self.data[self.session_id])
+        input_data = tc.SFrame(
+                {
+                    self.features[0]: [3.14],
+                    self.target: [sf_label],
+                    self.session_id: [sf_session_id],
+                }
+        )
+        model = tc.activity_classifier.create(
+                input_data,
+                features=["X1-r"],
+                target=self.target,
+                session_id=self.session_id,
+                prediction_window=self.prediction_window,
+        )
+        filename = tempfile.mkstemp("ActivityClassifier.mlmodel")[1]
```

Comment:
I remember that you are responsible to clear the temp file? how about using context like `with TempFile... as f`

Reply:
Oh good point!

> Unlike TemporaryFile(), the user of mkstemp() is responsible for deleting the temporary file when done with it.

 https://docs.python.org/3/library/tempfile.html

We do use  `tempfile.mkstemp` in a few places in the `test_activity_classifier.py` without clearing. Maybe I should do this for all places using it.

Reply:
I would just create a GitHub issues for this. I don't think we should worry about this problem at this point in the release cycle. I think it would be a good thing to fix early into the next release cycle.

Reply:
That sounds good. It makes sense to open another issue and track the usage across the repo and make proper file clearing logic where needed. Since we have this usage in a lot of places:

```
scenario-tests/tests/toolkits/test_image_classifier.py:        filename = tempfile.mkstemp('bingo.mlmodel')[1]
scenario-tests/tests/toolkits/test_image_classifier.py:        filename = tempfile.mkstemp('bingo.mlmodel')[1]
scenario-tests/tests/toolkits/test_image_similarity.py:        filename = tempfile.mkstemp('ImageSimilarity.mlmodel')[1]
src/python/turicreate/test/test_activity_classifier.py:        filename = tempfile.mkstemp("ActivityClassifier.mlmodel")[1]
src/python/turicreate/test/test_activity_classifier.py:        filename = tempfile.mkstemp("ActivityClassifier.mlmodel")[1]
src/python/turicreate/test/test_activity_classifier.py:                filename = tempfile.mkstemp("ActivityClassifier.mlmodel")[1]
src/python/turicreate/test/test_image_classifier.py:        filename = tempfile.mkstemp("bingo.mlmodel")[1]
src/python/turicreate/test/test_image_classifier.py:        filename = tempfile.mkstemp("bingo.mlmodel")[1]
src/python/turicreate/test/test_image_similarity.py:        filename = tempfile.mkstemp("ImageSimilarity.mlmodel")[1]
src/python/turicreate/test/test_object_detector.py:        filename = tempfile.mkstemp("bingo.mlmodel")[1]
src/python/turicreate/test/test_object_detector.py:        filename2 = tempfile.mkstemp("bingo2.mlmodel")[1]
src/python/turicreate/test/test_object_detector.py:        filename = tempfile.mkstemp("bingo.mlmodel")[1]
src/python/turicreate/test/test_object_detector.py:        filename2 = tempfile.mkstemp("bingo2.mlmodel")[1]
src/python/turicreate/test/test_one_shot_object_detector.py:        filename = tempfile.mkstemp("bingo.mlmodel")[1]
src/python/turicreate/test/test_one_shot_object_detector.py:        filename2 = tempfile.mkstemp("bingo2.mlmodel")[1]
src/python/turicreate/test/test_style_transfer.py:            filename = tempfile.mkstemp("my_style_transfer.mlmodel")[1]
src/python/turicreate/test/test_text_classifier.py:        filename = tempfile.mkstemp("bingo.mlmodel")[1]
src/python/turicreate/test/test_text_classifier.py:        filename = tempfile.mkstemp("bingo.mlmodel")[1]
```

Reply:
Opened an issue to track it https://github.com/apple/turicreate/issues/2993

Reply:
That's why we jam the test machines.